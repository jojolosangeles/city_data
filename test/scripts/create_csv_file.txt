# Load collision data into DataFrame "all_areas"
load ../data/real/Traffic_Collision_Data_from_2010_to_Present.csv as all_areas

# Display the columns (verify load)
columns all_areas

# Drop unused columns
drop all_areas DR Number, Date Reported, Area ID, Reporting District, Crime Code
drop all_areas Crime Code Description, MO Codes, Premise Code, Premise Description, Location

# Create columns for grouping
create all_areas Date [pd.to_datetime(x) for x in dataFrame['Date Occurred']]
create all_areas Weekday [x.weekday() for x in dataFrame['Date']]
create all_areas Year [x.year for x in dataFrame['Date']]
create all_areas Hour [math.floor(x/100) for x in dataFrame['Time Occurred']]

# Save DataFrame "all_areas" to CSV file "all_areas.csv"
save all_areas
