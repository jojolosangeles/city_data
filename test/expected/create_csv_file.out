Test: scripts/create_csv_file.txt
> # Load collision data into DataFrame "all_areas"
> load ../data/real/Traffic_Collision_Data_from_2010_to_Present.csv as all_areas
Loaded ../data/real/Traffic_Collision_Data_from_2010_to_Present.csv into DataFrame all_areas
> 
> # Display the columns (verify load)
> columns all_areas
['DR Number' 'Date Reported' 'Date Occurred' 'Time Occurred' 'Area ID'
 'Area Name' 'Reporting District' 'Crime Code' 'Crime Code Description'
 'MO Codes' 'Victim Age' 'Victim Sex' 'Victim Descent' 'Premise Code'
 'Premise Description' 'Address' 'Cross Street' 'Location']
> 
> # Drop unused columns
> drop all_areas DR Number, Date Reported, Area ID, Reporting District, Crime Code
Dropping ['DR Number', 'Date Reported', 'Area ID', 'Reporting District', 'Crime Code'] from DataFrame 'all_areas'
Columns dropped
> drop all_areas Crime Code Description, MO Codes, Premise Code, Premise Description, Location
Dropping ['Crime Code Description', 'MO Codes', 'Premise Code', 'Premise Description', 'Location'] from DataFrame 'all_areas'
Columns dropped
> 
> # Create columns for grouping
> create all_areas Date [pd.to_datetime(x) for x in dataFrame['Date Occurred']]
all_areas.Date=
> create all_areas Weekday [x.weekday() for x in dataFrame['Date']]
all_areas.Weekday=
> create all_areas Year [x.year for x in dataFrame['Date']]
all_areas.Year=
> create all_areas Hour [math.floor(x/100) for x in dataFrame['Time Occurred']]
all_areas.Hour=
> 
> # Save DataFrame "all_areas" to CSV file "all_areas.csv"
> save all_areas
Saved CSV File 'all_areas.csv'
Test completed: scripts/create_csv_file.txt
